pkgs/games/infra-arcana/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, cmake
+, makeWrapper
+, SDL2
+, SDL2_image
+, SDL2_mixer
+}:
+
+stdenv.mkDerivation rec {
+  pname = "infra-arcana";
+  version = "21.0.1";
+
+  src = fetchFromGitLab {
+    owner = "martin-tornqvist";
+    repo = "ia";
+    rev = "v${version}";
+    sha256 = "sha256-E2ssxdYa27qRk5cCmM7A5VqXGExwXHblR34y+rOUBRI=";
+  };
+
+  nativeBuildInputs = [ cmake makeWrapper ];
+  buildInputs = [ SDL2 SDL2_image SDL2_mixer ];
+
+  # Need to use a newer version of the Catch2 header file to get a
+  # glibc 2.34-related fix.
+  patches = [ ./updateCatch2.patch ];
```

Comment:
Can we upstream this or do something different to not vendor a 13000 lines file?

Reply:
I just realized this too, and was asking in Matrix about it. I'm definitely trying to upstream the version update, as they're using a pretty outdated one. I think I should better put this on hold until it's upstreamed.

Could be that it takes a while, as the tests in their `develop` branch are currently broken. I'd hope that they get fixed and I'd  update this PR with the development version then.